### PR TITLE
Resolves #533: New parameter to DisplayPluginUpdatesMojo: processUnboundPlugins

### LINKED ISSUE #533: display-plugin-updates should optionally show plugins updates for unbound plugins
Currently the goal `display-plugin-updates` only shows updates for plugins bound in the current project's lifecycle and plugins listed in the effective model in build->plugins->plugin (see https://github.com/mojohaus/versions-maven-plugin/blob/c0589f4a69afc1a53d8f2d461f621339fd0c0710/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java#L380).
In some cases one is really interested to see all possible plugin updates, even for the ones which are neither bound (e.g. because the project is of type "pom") nor enabled or ones only listed in some profiles.

A common use case is to check a parent pom for possible plugin updates. That will currently never emit infos for updates of e.g. "maven-jar-plugin" unless that plugin is explicitly mentioned (due to its packaging "pom").

### LINKED ISSUE #550: display-plugin-updates: Inherited plugin version not detected
When cloning https://github.com/apache/sling-parent and executing ` mvn org.codehaus.mojo:versions-maven-plugin:2.9.0:display-plugin-updates` it emits

```
[WARNING] The following plugins do not have their version specified:
[WARNING]   maven-jar-plugin ......................................... 3.2.2
```

for the module `sling-bundle-parent`, although the version of this plugin is managed in https://github.com/apache/sling-parent/blob/b7104da953b090821549d23f9bdf1ea1d002a1d8/sling-parent/pom.xml#L249 (the parent).

### PR DESCRIPTION
**Pull request by ajarmoniuk**
- if true, will show updates to plugins defined in parent poms, which have no version in the current pom
- false by default to retain backwards compatibility

@slawekjaranowski please review

### CODE REVIEW

**Review by slachiewicz — APPROVED** [review]

File: versions-maven-plugin/src/it/it-display-plugin-updates-issue-533/invoker.properties [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/it/it-display-plugin-updates-issue-533/invoker.properties:2]: New file please use standard header

File: versions-maven-plugin/src/it/it-display-plugin-updates-issue-533/parent.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/it/it-display-plugin-updates-issue-533/parent.xml:2]: New file please use standard header

File: versions-maven-plugin/src/it/it-display-plugin-updates-issue-533/parent.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>localhost</groupId>
+  <artifactId>it-display-plugin-updates-issue-550-parent</artifactId>
+  <version>1.0</version>
+  <packaging>pom</packaging>
+
+  <prerequisites>
+    <maven>${mavenVersion}</maven>
+  </prerequisites>
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/it/it-display-plugin-updates-issue-533/parent.xml:31]: Should be used only for maven-plugin packaging

File: versions-maven-plugin/src/it/it-display-plugin-updates-issue-533/verify.groovy [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/it/it-display-plugin-updates-issue-533/verify.groovy:2]: not asf

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**